### DISCUSSION TITLE
Will S3TransferManager from java SDK 2 abort Multipart uploading process in case of unchecked Exceptions?

### DISCUSSION THREAD

**Discussion by vlbaluk** [aws/aws-sdk-java-v2#4882] [Migration from v1 to v2]
We are migrating from Sync Client to Async Crt Client and S3TransferManager. 

I didn't find in the documentation how S3TransferManager behaves in case of any unchecked exception.
Does S3TransferManager from Java SDK 2 abort the Multipart uploading process and automatically remove all already uploaded chunks in case of unchecked Exceptions?
Found thread where the desired behavior is described, but only in case the system explicitly cancels `CompletableFuture`
https://github.com/aws/aws-sdk-java-v2/issues/3274#issuecomment-1182442921

Our code listing : 

    final Path tempFile = createTempFile(S3Path.class.getSimpleName(), null);
        try {
                Files.copy(source, tempFile, REPLACE_EXISTING);
                UploadFileRequest uploadFileRequest = UploadFileRequest.builder()
                .putObjectRequest(b -> b.bucket(bucketName).key(getS3Key()))
                .source(Paths.get(tempFile.toUri()))
                .build();
                FileUpload fileUpload = s3TransferManager.uploadFile(uploadFileRequest);
                Object res =  fileUpload.completionFuture().join().response();
        } finally {
            Files.delete(tempFile);
        }`

**Comment by debora-ito** [comment]
I've been told that yes, S3TransferManager will abort the multipart upload in case of an exception.
